sec1_intro_and_setup/test_app/app/assets/stylesheets/application.css

```diff
@@ -0,0 +1,15 @@
+/*
```

Comment:
Rails needs to know which files to include in that giant blob, so it uses manifest files to determine this. 
(Note that: Binary Large OBject (BLOB) is a collection of binary data stored as a single entity. Blobs are typically images, audio or other multimedia objects, though sometimes binary executable code is stored as a blob.)

The `*= require_tree .` syntax is responsible for rendering all the files within `app/assets/stylesheets` and automatically including them here so that developer never have to include them manually. The `*= require_self` comment requires the file itself which is what ends up in our main layout template. You can add any CSS file you like and require it here if it is not already inside the `app/assets/stylesheets` directory.

Reply:
> in that giant blob

1. What is this referring to?
2. Explain to me this syntax `*= require_tree .` How does it equal to rendering all files within `app/assets/stylesheets`?

*The best way to learn is to read, understand, then use your own words. Not copy paste from the source.

Reply:
1. Rails concatenate all your asset files together into one big one for each file type. The giant blob refers to that big file type.
2. The `require_tree` grabs everything in the current directory, so it brings in all CSS files in the current directory as stated in the comment in this file.
> Any CSS and SCSS file within this directory, lib/assets/stylesheets, vendor/assets/stylesheets, or vendor/assets/stylesheets of plugins, if any, can be referenced here using a relative path.

Reply:
1. Why rails does this?
2. 100% wrong

Reply:
1. To reduce the number of files and the size of files. Reducing the number of files allows the browser to make fewer HTTP requests. Reducing the size of asset files can reduce the number of bytes to send over the network. Both of these features allow the browser to load pages faster.
2. `require_tree` is used to tell the asset pipeline to include all files in the specified directory.

Reply:
> To reduce the number of files and the size of fi...

very good

> require_tree is used to tell the asset pipeline to include all files in the specified directory.

wrong

Reply:
The `require_tree` in this file requires all stylesheets from the current directory.

Reply:
> The require_tree in this file requires all stylesheets from the current directory.

**WRONG**

Reply:
`require_tree .` where `.` represents the current directory while `require_tree` requires files in the subdirectories. This means that every CSS in the current directory and subdirectory of `app/assets/stylesheets` will be included since the Asset Pipeline centralises all the CSS files in `app/assets/stylesheets`.

Reply:
hello.

Instead of explaining to me whats what. See properly 2,3,4,5,6 times what you mention that will require all stylesheet from the current directory. Its not the same as what you mentioned in this last comment

Reply:
In `app/views/layouts/application.html.erb`, there is a line `<%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>` where `application` refers to including the stylesheets called `application.css`(which is this file). Rails will use the layout defined in the file `app/views/layouts/application.html.erb` as a default for rendering any page. Since `require_tree .` is in `application.css` which requires all stylesheets from the current directory, so the `require_tree .` is equal to rendering all files within `app/assets/stylesheet`.

Reply:
Irrelevant

Reply:
Lets not waste time anymore.

![image](https://user-images.githubusercontent.com/5483026/130703217-104904c8-5a76-40db-991c-594f794f4328.png)

You tell me `require_tree` will require all the stylesheets from the current directory.
This is *WRONG*.

The correct statement is:
`require_tree .` will require all the stylesheets from the current directory.